pkg/cli/clients/clients.go

```diff
@@ -227,8 +227,17 @@ type ApplicationsManagementClient interface {
 	// GetResourceProviderSummary gets the resource provider summary with the specified name in the configured scope.
 	GetResourceProviderSummary(ctx context.Context, planeName string, providerNamespace string) (ucp_v20231001preview.ResourceProviderSummary, error)
 
+	// CreateOrUpdateResourceType creates or updates a resource type in the configured scope.
+	CreateOrUpdateResourceType(ctx context.Context, planeName string, providerNamespace string, resourceTypeName string, resource *ucp_v20231001preview.ResourceTypeResource) (ucp_v20231001preview.ResourceTypeResource, error)
+
 	// DeleteResourceType deletes a resource type in the configured scope.
 	DeleteResourceType(ctx context.Context, planeName string, providerNamespace string, resourceTypeName string) (bool, error)
+
+	// CreateOrUpdateAPIVersion creates or updates an API version in the configured scope.
+	CreateOrUpdateAPIVersion(ctx context.Context, planeName string, providerNamespace string, resourceTypeName string, apiVersionName string, resource *ucp_v20231001preview.APIVersionResource) (ucp_v20231001preview.APIVersionResource, error)
```

Comment:
So are you suggesting a name like `CreateOrUpdateResourceTypeAPIVersion`?

I don't have a strong opinion about whether that's better 😆 

My overall opinion is that this client type is pretty unwieldy and needs a redesign because maintaining it is hard. I don't want to do that in the PR.

Reply:
sounds good

Reply:
After thinking about this, I decided not to apply this feedback. The functions here are consistent with the naming of the generated APIs and the OpenAPI/TypeSpec. Trying to fight that is going to introduce more confusion.

If we encounter an case where the concepts are actually overloaded, we'll address it then.